src/services/utm-cookies-manager.test.js

```diff
@@ -1,28 +1,44 @@
 import { UtmCookiesManager } from './utm-cookies-manager';
 import { FakeLocalStorage } from './test-utils/local-storage-double';
 
+const saveMockDataInStorage = (utmCookiesManager, storage) =>
+  utmCookiesManager.setCookieEntry({
+    storage,
+    UTMSource: 'source',
+    UTMCampaign: 'campaign',
+    UTMMedium: 'medium',
+    UTMTerm: 'term',
+    gclid: 'gclid',
+  });
+
 describe('Utm cookies manager', () => {
-  it('should initialize add only an entry per instance', () => {
+  let utmCookiesManager;
+  let storage;
+
+  beforeEach(() => {
     // Arrange
-    const utmCookiesManager = new UtmCookiesManager();
-    const storage = new FakeLocalStorage();
+    utmCookiesManager = new UtmCookiesManager();
+    storage = new FakeLocalStorage();
+  });
+
+  it('should initialize add only an entry per instance', () => {
     // Act
-    utmCookiesManager.setCookieEntry(storage, 'source', 'campaign', 'medium', 'term');
-    utmCookiesManager.setCookieEntry(storage, 'source', 'campaign', 'medium', 'term');
+    saveMockDataInStorage(utmCookiesManager, storage);
+    saveMockDataInStorage(utmCookiesManager, storage);
+
     // Assert
     expect(utmCookiesManager.getUtmCookie(storage).length).toBe(1);
   });
 
   it('should add utms into utmCookie', () => {
-    // Arrange
-    const utmCookiesManager = new UtmCookiesManager();
-    const storage = new FakeLocalStorage();
     // Act
-    utmCookiesManager.setCookieEntry(storage, 'source', 'campaign', 'medium', 'term');
+    saveMockDataInStorage(utmCookiesManager, storage);
+
     // Assert
     expect(utmCookiesManager.getUtmCookie(storage)[0].UTMSource).toBe('source');
     expect(utmCookiesManager.getUtmCookie(storage)[0].UTMCampaign).toBe('campaign');
     expect(utmCookiesManager.getUtmCookie(storage)[0].UTMMedium).toBe('medium');
     expect(utmCookiesManager.getUtmCookie(storage)[0].UTMTerm).toBe('term');
+    expect(utmCookiesManager.getUtmCookie(storage)[0].gclid).toBe('gclid');
```

Comment:
@jcamposmk could you add a tests for missing gclid or other utm values?

Reply:
I already did it!